docs/style.md

```diff
@@ -0,0 +1,188 @@
+
+Avoid: `controls/ssh_controls.rb`
+Use: `controls/ssh.rb`
+
+
+## Code style
+
+### 3. Use Ruby style conventions
+
+Reason: There are many opinions on how to style code across languages and frameworks. To reduce the differences and make all code more easily readable and understandable we choose to follow some common Ruby style guidelines. The following list is a subset of the current best practices. Since InSpec is focused on writing tests instead of arbitrary programming code we can focus on a subset that is important to test authors.
+
+This is a subset of style conventions taken from [the Ruby style guide](https://github.com/rubocop-hq/ruby-style-guide):
+
+- Use: UTF-8 as source file encoding.
+- Use: 2 spaces per indentation level. No hard tabs
+- Use: Unix-style file endings.
+- Avoid: multiple expressions per line.
+- Use: Spaces around operators, after commas, colons and semicolons.
+```ruby
+sum = 1 + 2
+```
+- Avoid: Spaces after `(`, `[` or before `]`, `)`. Use spaces around `{` and before `}`.
+```ruby
+# bad
+some( arg ).other
+[ 1, 2, 3 ].each{|e| puts e}
+
+# good
+some(arg).other
+[1, 2, 3].each { |e| puts e }
+```
+- Avoid: several empty lines in a row
+
+Please refer to the full guide for the recommended style conventions.
+
+## Controls
+
+### 4. Do not wrap controls in conditional statements
+
+Reason: This will create dynamic profiles whose controls depend on the execution. The problem here is that we cannot render the profile or provide its information before scanning a system. We want to be able to inform users of the contents of their profiles before they run them. It is valid to skip controls that are not necessary for a system, as long as you do it via `only_if` conditions. Ruby's internal conditionals will hide parts of the profile to static analysis and should thus be avoided.
+
+Avoid:
+```ruby
+if package('..').installed?
+  control "package-test1" do
+    ..
+  end
+end
+```
+
+Use:
+```ruby
+control "package-test1" do
+  only_if { package('..').installed? }
+end
+```
+
+Avoid:
+```ruby
+case inspec.platform.name
+when /centos/
+  include_controls 'centos-profile'
+...
+```
+Use:
+- Set the `supports` attribute in `inspec.yml` files of the profile you want to include:
+```ruby
+supports:
+- platform-name: centos
+```
+Now whenever you run the base profile you can just `include_controls 'centos-profile'`.
+It will only run the included profiles is the platform matches the supported platform.
+
+
+### 5. Do not include dynamic elements in the control IDs
+
+Reason: Control IDs are used to map test results to the tests and profiles. Dynamic control IDs make it impossible to map results back, since the identifier which connects tests and results may change in the process.
+
+Avoid:
+```ruby
+control "test-file-#{name}" do
+  ..
+end
+```
+
+Use:
+```ruby
+control "test-all-files" do
```

Comment:
Not sure how we stand on single vs double quotes...

Reply:
That's a really good question and discussion point! Opinions?

Reply:
(the ruby style guide is opinionated on this; the question is: do we put our users through this?)

Reply:
I am the worst person to ask on this :wink:. I try to write all my code to stock rubocop...but wouldn't wish that on anyone else.

Reply:
Answered in parent thread. TL;DR: I care too much :smile: